File: day1/day_1.rb
require 'pry'

file = File.open("input.txt")
file_data = file.read
elves = file_data.split("\n\n")
elves_with_numbers = elves.map do |elf|
  elf.split("\n").map{ |number| number.to_i }
end

elves_with_numbers = elves_with_numbers.map(&:sum).sort.reverse

puts elves_with_numbers[0..2].sum

File: elixir/day1/day1.exs
defmodule Day1 do
    def execute() do
            with {:ok, file} <- File.read("input.txt"),
            elves <- String.split(file,"\n\n") do
                IO.puts(Enum.at(elves, 0))
                elves |>
                Enum.map(&convert_to_sum/1) |>
                Enum.max |>
                IO.puts
        else
        {:error, _} ->
            IO.puts("everything went wrong")
        end
    end

    def convert_to_sum(string) do
        string |>
        String.split("\n") |>
        Enum.map(&String.to_integer/1) |>
        Enum.sum
    end
end

Day1.execute()
